### DISCUSSION TITLE
Promote touchplate from wire extras

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#2639] [Suggestions]
A button that responds to touch/force.

In essence, like a physical prop "button" made with a hydraulic, but in Lua form. A player, prop, or some other physical trigger will activate the button by touching it or applying force (how could this be measured?) to it. Maybe a specific amount of force has to be applied to trigger the button. An alternative idea could simply be a "force detector", or on the easier end of the spectrum, a "touch detector" (so just a trigger?).

Only issue is that I don't know if anything like that would be possible.

**Comment by thegrb93** [comment]
Take a look at https://github.com/thegrb93/StarfallEx/blob/a64078e0930a9e4b53974ff30213d88138d63fe0/lua/starfall/libs_sv/entities.lua#L311-L325

**Comment by Denneisk** [comment]
Oh, awesome! I was really hoping something like that would exist.
Maybe I'll make a prototype of this sometime.

**Comment by vurvdev** [comment]
So a pressure plate

**Comment by vurvdev** [comment]
This already exists actually, in wire extras

https://github.com/wiremod/wire-extras/tree/master/lua/entities/gmod_wire_touchplate

**Comment by vurvdev** [comment]
Could be brought into official wire maybe with a little bit of refactoring.

If we want the weight functionality that wouldn't be too hard. Just add up the forces/weights of all the entities onto the prop and have a minimum force to trigger it.

**Comment by vurvdev** [comment]
So closing this unless you want to turn this issue into discussion for promoting the touch plate/pressure pad into regular wiremod.

**Comment by Denneisk** [comment]
Well, yes, that was the idea.

**Comment by thegrb93** [comment]
If the license of both repo are the same then its trivial. Its just a matter of whether it's in working condition.
